packages/kit/src/runtime/server/index.js

```diff
@@ -260,7 +261,11 @@ export async function respond(request, options, state) {
 												/** @type {import('types').JSONObject} */
 												const data = {};
 												for (let j = 0; j < i; j += 1) {
-													Object.assign(data, await promises[j]);
+													const parent = await promises[j];
+													if (!parent || parent instanceof HttpError || 'error' in parent) {
+														continue;
```

Comment:
no point going any further, we can just bail at this point

```suggestion
														return;
```